File: app/src/main/java/com/supercilex/robotscouter/ui/scout/viewholder/template/SpinnerItemViewHolder.java
package com.supercilex.robotscouter.ui.scout.viewholder.template;

import android.support.annotation.Keep;
import android.support.v7.widget.RecyclerView;
import android.text.TextUtils;
import android.view.View;
import android.widget.EditText;

import com.google.firebase.database.DataSnapshot;
import com.supercilex.robotscouter.R;

public class SpinnerItemViewHolder extends RecyclerView.ViewHolder implements ScoutTemplateViewHolder {
    private String mPrevText;

    private EditText mItemEditText;
    private DataSnapshot mSnapshot;

    @Keep
    public SpinnerItemViewHolder(View itemView) {
        super(itemView);
        mItemEditText = (EditText) itemView.findViewById(R.id.name);
    }

    public void bind(String text, DataSnapshot snapshot) {
        mPrevText = text;
        mSnapshot = snapshot;

        mItemEditText.setText(text);
        mItemEditText.setOnFocusChangeListener(this);
    }

    @Override
    public void requestFocus() {
        mItemEditText.requestFocus();
    }

    @Override
    public void onFocusChange(View v, boolean hasFocus) {
        String text = mItemEditText.getText().toString();
        if (!hasFocus && !TextUtils.equals(text, mPrevText)) {
            mSnapshot.getRef().setValue(text, mSnapshot.getPriority());
        }
    }
}

File: app/src/main/java/com/supercilex/robotscouter/ui/scout/viewholder/template/SpinnerItemViewHolder.kt
package com.supercilex.robotscouter.ui.scout.viewholder.template

import android.support.annotation.Keep
import android.support.v7.widget.RecyclerView
import android.text.TextUtils
import android.view.View
import android.widget.EditText

import com.google.firebase.database.DataSnapshot
import com.supercilex.robotscouter.R

class SpinnerItemViewHolder @Keep constructor(itemView: View) :
        RecyclerView.ViewHolder(itemView), ScoutTemplateViewHolder {
    private val itemEditText: EditText = itemView.findViewById(R.id.name) as EditText

    private lateinit var prevText: String
    private lateinit var snapshot: DataSnapshot

    fun bind(text: String, snapshot: DataSnapshot) {
        prevText = text
        this.snapshot = snapshot

        itemEditText.setText(text)
        itemEditText.onFocusChangeListener = this
    }

    override fun requestFocus() {
        itemEditText.requestFocus()
    }

    override fun onFocusChange(v: View, hasFocus: Boolean) {
        val text: String = itemEditText.text.toString()
        if (!hasFocus && !TextUtils.equals(text, prevText)) {
            snapshot.ref.setValue(text, snapshot.priority)
        }
    }
}
